### DISCUSSION TITLE
VM timeout is not working

### DISCUSSION THREAD

**Discussion by LijieZhang1998** [nodejs/help#4779] [Q&A]
Hi, I'm attempting to interrupt python code using pyodide running inside of vm. The `python while` loop can't be interrupted. Is there any idea? Thanks.

This is my code.
```
(async function () {
  const vm = require("node:vm");
  const { loadPyodide } = require("pyodide");
  let pyodide = await loadPyodide();
  const src = `
while True:
    print('in the python while loop')
print('The python while loop is done')
        `;

  const code = `
  pyodide.runPythonAsync(src).then(()=>{ console.log("python is done")});
    `;
  vm.runInNewContext(
    code,
    { console, pyodide, src },
    { timeout: 1000, microtaskMode: "afterEvaluate" }
  );
  console.log("done");
})();

**Comment by bnoordhuis** [comment]
pyodide is a WASM blob and I'm like 85% sure WASM isn't interruptible like JS code is. You could try reporting that to the V8 project if there isn't an open bug report about it already.

Technical details: I don't think V8's WASM compiler emits the StackCheck nodes that are necessary to detect execution termination. More specifically, leaf functions that don't throw or grow memory don't get them, and pyodide's main interpreter loop probably is that kind of leaf function.
